spi/src/main/java/org/eclipse/dataspaceconnector/spi/types/domain/transfer/DataRequest.java

```diff
@@ -38,7 +38,7 @@
 
     private String connectorId;
 
-    private DataEntry dataEntry;
+    private Asset asset;
```

Comment:
how about using the Asset-ID (i.e. a String)? Not much use in passing the whole asset object here. The only usage I could find was in `DataRequestMessageSender` and that only uses the Asset ID